source/extensions/filters/network/ext_authz/ext_authz.cc

```diff
@@ -53,6 +68,16 @@ Network::FilterStatus Filter::onData(Buffer::Instance&, bool /* end_stream */) {
 }
 
 Network::FilterStatus Filter::onNewConnection() {
+  if (config_->checkOnNewConnection()) {
```

Comment:
I think it should be either filter state or `StopIteration` - but not both to make `tcp_proxy` from opening an upstream connection, based on the documentation.

Reply:
Yes, you are right. I was under the impression that the `onNewConnection` of any downstream filters would not be called if reads were disabled. Removed all changes related to the filter state.